pkg/background/update_request_controller.go

```diff
@@ -241,6 +245,34 @@ func (c *controller) syncUpdateRequest(key string) error {
 	return err
 }
 
+func (c *controller) checkIfCleanupRequired(ur *kyvernov1beta1.UpdateRequest) error {
+	var err error
+	pNamespace, pName, err := cache.SplitMetaNamespaceKey(ur.Spec.Policy)
+	if err != nil {
+		return err
+	}
+
+	if pNamespace == "" {
+		_, err = c.cpolLister.Get(pName)
+	} else {
+		_, err = c.polLister.Policies(pNamespace).Get(pName)
+	}
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+		statusControl := common.NewStatusControl(c.kyvernoClient, c.urLister)
+		ctrl := generate.NewGenerateController(c.client, c.kyvernoClient, statusControl, c.cpolLister, c.polLister, c.urLister, c.nsLister, c.configuration, c.eventGen, logger)
```

Comment:
Creating a new instance of the controller doesn't look right. We should update the UR status to trigger a reconciliation.

Reply:
Right , i tried but update the status to pending to trigger reconciliation, but that will not work b/c there will be a race condition. Let me carve out the cleanup functions and fix this.